providers/asrockrack/asrockrack.go

```diff
@@ -148,7 +148,8 @@ func (a *ASRockRack) GetBMCVersion(ctx context.Context) (string, error) {
 // nolint: gocyclo
 // BMC firmware update is a multi step process
 // this method initiates the upgrade process and waits in a loop until the device has been upgraded
-func (a *ASRockRack) FirmwareUpdateBMC(ctx context.Context, fileReader io.Reader) error {
+// fileSize is required to setup the multipart form upload Content-Length
+func (a *ASRockRack) FirmwareUpdateBMC(ctx context.Context, fileReader io.Reader, fileSize int64) error {
```

Comment:
Just as a general process comment, since the two things mentioned in the commit message here are as far as I can tell completely independent, it'd preferable to have them in separate commits.  (Makes review easier, and fine-grained git history is more useful for things like potential future reverts and bisecting.)